### PR TITLE
feat: outofband BDD SDK tests

### LINKED ISSUE #1531: BDD test for out-of-band + didexchange : request message
part of #1454

This will be a simple BDD test to trigger a didexchange flow from an out-of-band `request` message.

Note: in this initial test the `request` message will not have embedded attachments although it should (see hyperledger/aries-rfcs#451). A followup task will refactor this test to trigger an additional protocol after did-exchange is complete.

### LINKED ISSUE #1552: Error BDD tests: "too many open files"
Running BDD tests locally results in this error:

<details><summary>expand</summary>

```
panic: error issuing command to docker-compose with args '[logs]':  pipe2: too many open files ()

goroutine 1 [running]:
github.com/hyperledger/aries-framework-go/test/bdd.runBddTests.func1.2()
        /home/llorllale/dev/hyperledger/aries-framework-go/test/bdd/bddtests_test.go:109 +0x23a
github.com/cucumber/godog.(*Suite).run(0xc000453680)
        /home/llorllale/go/pkg/mod/github.com/cucumber/godog@v0.8.1/suite.go:275 +0x190
github.com/cucumber/godog.(*runner).run(0xc00045ea80, 0x14)
        /home/llorllale/go/pkg/mod/github.com/cucumber/godog@v0.8.1/run.go:118 +0x25c
github.com/cucumber/godog.RunWithOptions(0x120db46, 0x6, 0x1373e78, 0x0, 0x1601d015ba71069c, 0x101, 0x120c312, 0x3, 0x120dc5a, 0x6, ...)
        /home/llorllale/go/pkg/mod/github.com/cucumber/godog@v0.8.1/run.go:215 +0xbc2
github.com/hyperledger/aries-framework-go/test/bdd.runBddTests(0x120c312, 0x3, 0x120dc5a, 0x6, 0x0)
        /home/llorllale/dev/hyperledger/aries-framework-go/test/bdd/bddtests_test.go:78 +0x2c9
github.com/hyperledger/aries-framework-go/test/bdd.TestMain(0xc000218880)
        /home/llorllale/dev/hyperledger/aries-framework-go/test/bdd/bddtests_test.go:68 +0x1c4
main.main()
        _testmain.go:82 +0x334
FAIL    github.com/hyperledger/aries-framework-go/test/bdd      17.533s
```
</details>

**Problem:** @soluchok pointed out that BDD tests are not doing proper cleanup. Some bdd test packages have started doing this cleanup individually (eg. `introduce`, `outofband`).

It would be great if this cleanup was done automatically using one of godog's hooks (eg. `AfterScenario`).

Things to watch out for:

* `BDDContext.AgentCtx` holds `context.Provider` which have no `Close()` method. Should be refactored to hold `framework.Aries` instead
* [Concurrency](https://github.com/cucumber/godog/issues/180)

### PR DESCRIPTION
**Pull request by llorllale**
Closes #1531 
Closes #1552

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: scripts/check_go_integration.sh [hunk 1]
```diff
@@ -9,7 +9,8 @@ set -e
 echo "Running aries-framework-go integration tests..."
 PWD=`pwd`
 cd test/bdd
-go test -count=1 -v -cover . -p 1 -timeout=20m -race
+#go test -count=1 -v -cover . -p 1 -timeout=20m -race
```
Comment (troyronda) [inline scripts/check_go_integration.sh:12]: leftover.

**Comment by llorllale** [comment]
The BDD tests timed out in the original run: https://github.com/hyperledger/aries-framework-go/runs/553404684?check_suite_focus=true

I temporarily disabled all but the outofband_e2e_sdk tests and can confirm they work in the CI: https://github.com/hyperledger/aries-framework-go/pull/1553/checks?check_run_id=555061443

Need to figure out what to do w.r.t. the timeout.

File: scripts/check_go_integration.sh (hunk 1 above)
Comment (llorllale) [reply scripts/check_go_integration.sh]: it was temporary - see https://github.com/hyperledger/aries-framework-go/pull/1553#issuecomment-607845970

**Comment by llorllale** [comment]
Fixing #1552 also seems to fix the timeout issue I was having before.

**Pull request merged**